sql/core/src/main/scala/org/apache/spark/sql/execution/LocalTableScanExec.scala

```diff
@@ -49,7 +49,9 @@ case class LocalTableScanExec(
     if (rows.isEmpty) {
       sqlContext.sparkContext.emptyRDD
     } else {
-      val numSlices = math.min(unsafeRows.length, sqlContext.sparkContext.defaultParallelism)
+      val numSlices = math.min(
+        unsafeRows.length,
+        conf.filesMinPartitionNum.getOrElse(sqlContext.sparkContext.defaultParallelism))
```

Comment:
@wangyum, I think this configuration `spark.sql.files.minPartitionNum` is only for files as the naming shows. It looks a bit odd to apply this in local collection.

Reply:
Yes, I also think it’s a little bit odd. What do you think of setting numSlices to 1?

Reply:
hm, how about adding a new config to control each partition size? I'm not sure about why `numSlice` depends on the length of `unsafeRows`.